### PR TITLE
rc4.1: ignore calculate regions size when heartbeat to pd

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Get approximate size may block raftstore thread, move it to pd worker.
cherry-pick https://github.com/pingcap/tikv/commit/df6fcae6efcf0231f8523e5c1ce100c445c80270

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

But seem we still need this to let PD do balance more accurately because the region size may be different later.

**Comment by siddontang** [comment]
@zhangjinpeng1987 

How long will the getting approximate size take?

File: src/raftstore/store/peer.rs [hunk 1]
```diff
@@ -1547,7 +1547,7 @@ impl Peer {
             pending_peers: self.collect_pending_peers(),
             written_bytes: self.peer_stat.last_written_bytes,
             written_keys: self.peer_stat.last_written_keys,
-            approximate_size: self.approximate_size().unwrap_or(0),
```
Comment (ngaut) [inline src/raftstore/store/peer.rs]: Do we still need to keep approximate_size field?

Comment (zhangjinpeng1987) [reply src/raftstore/store/peer.rs]: done

**Pull request merged**